pubsub/gossipsub/gossipsub-v1.2.md

```diff
@@ -0,0 +1,105 @@
+# Gossipsub v1.2
+
+# Overview
+
+This document aims to provide a minimal extension to the [gossipsub
+v1.1](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md)
+protocol.
+
+The proposed extensions are backwards-compatible and aim to add utility control messages for 
+better network efficiency (delivery latency and total throughput) tuning of gossip protocol 
+implementations 
+
+In more specific terms, new control messages are introduced: `PING` and `PONG`. 
+They are primarily intended for collecting statistics to estimate peer connection 
+latency and bandwidth and adjusting message broadcasting strategy accordingly.
+
+# Specification
+
+## Protocol Id
+
+Nodes that support this Gossipsub extension should additionally advertise the
+version number `1.2.0`. Gossipsub nodes can advertise their own protocol-id
+prefix, by default this is `meshsub` giving the default protocol id:
+- `/meshsub/1.2.0`
+
+## Parameters
+
+This section lists the configuration parameters that needs to agreed on across clients to avoid
+peer penalizations
+
+| Parameter         | Description                                                            | Reasonable Default |
+|-------------------|------------------------------------------------------------------------|--------------------|
+| `min_ping_period` | minimal delay between sending next `PING` and previous received `PONG` | 1 second (?)       |
+| `max_ping_rate`   | (better alternative) maximum number of `PING` messages per minute      | 60                 |
```

Comment:
max_ping_rate does seem better, but 60/min is far too low
If we ping every other RTT with 20ms ping, we are already at 25ping/s

Reply:
Ok will leave `max_ping_rate `
>  but 60/min is far too low

Yep, but this a 'reasonable default' for normal Libp2p usage 😆 
We will adjust this option for Eth2 when we need